common/kafka/stats_enum.cpp

```diff
@@ -25,42 +25,7 @@ DEFINE_string(index_type, "", "The index type");
 DEFINE_string(normalized_cluster_name, "", "The normalized cluster name");
 DEFINE_string(stats_prefix, "", "Prefix for the stats");
 
-const std::vector<std::string> kCounterNames = {
-#define NEW_COUNTER_STAT(a, b) b,
-    NEW_COUNTER_STAT(kKafkaConsumerSeek, "kafka_consumer_seek")
-    NEW_COUNTER_STAT(kKafkaConsumerErrorInit, "kafka_consumer_error_init")
-    NEW_COUNTER_STAT(kKafkaConsumerErrorAssign, "kafka_consumer_error_assign")
-    NEW_COUNTER_STAT(kKafkaConsumerErrorSeek, "kafka_consumer_error_seek")
-    NEW_COUNTER_STAT(kKafkaConsumerErrorConsume, "kafka_consumer_error_consume")
-    NEW_COUNTER_STAT(kKafkaConsumerMessageNull, "kafka_consumer_message_null")
-    NEW_COUNTER_STAT(kKafkaWatcherMessageMissing,
-        "kafka_watcher_message_missing")
-    NEW_COUNTER_STAT(kKafkaWatcherBlockingConsumeTimeout,
-        "kafka_watcher_blocking_consume_timeout")
-#undef NEW_COUNTER_STAT
-};
-
-const std::vector<std::string> kMetricNames = {
-#define NEW_COUNTER_STAT(a, b)
-    NEW_COUNTER_STAT(kKafkaConsumerSeek, "kafka_consumer_seek")
-    NEW_COUNTER_STAT(kKafkaConsumerErrorInit, "kafka_consumer_error_init")
-    NEW_COUNTER_STAT(kKafkaConsumerErrorAssign, "kafka_consumer_error_assign")
-    NEW_COUNTER_STAT(kKafkaConsumerErrorSeek, "kafka_consumer_error_seek")
-    NEW_COUNTER_STAT(kKafkaConsumerErrorConsume, "kafka_consumer_error_consume")
-    NEW_COUNTER_STAT(kKafkaConsumerMessageNull, "kafka_consumer_message_null")
-    NEW_COUNTER_STAT(kKafkaWatcherMessageMissing,
-        "kafka_watcher_message_missing")
-    NEW_COUNTER_STAT(kKafkaWatcherBlockingConsumeTimeout,
-        "kafka_watcher_blocking_consume_timeout")
-#undef NEW_COUNTER_STAT
-};
-
-std::string getFullCounterName(const CounterIdx idx,
-                               const std::initializer_list<std::string>& tags) {
-  return getFullMetricName(kCounterNames[idx], tags);
-}
-
-std::string getFullMetricName(const std::string& metric_name,
```

Comment:
Not sure how frequently this function will be called.
We may want to optimize it by call a std::string::reserve() on full_metric_name. We can do it in a separate diff to minimize the change to muse.

Reply:
It will be called very frequently, will do as you suggest in a separate diff.